src/program/lwaftr/run/run.lua

```diff
@@ -136,9 +136,14 @@ end
 -- Requires a V4V6 splitter if running in on-a-stick mode and VLAN tag values
 -- are the same for the internal and external interfaces.
 local function requires_splitter (opts, conf)
+   local function first (t)
+      assert(type(t) == "table")
+      for k, v in pairs(t) do return v end
+   end
    if opts["on-a-stick"] then
-      local internal_interface = conf.softwire_config.internal_interface
-      local external_interface = conf.softwire_config.external_interface
+      local queue = first(conf.softwire_config.instance).queue
```

Comment:
Throughout the code the way I've done this instead of using the `first` function is just to extract the first one using `next`. It might be worth either switching to that or changing the other instances to this for consistency sake. It's not a huge thing though and it doesn't have to change, I'm just being a bit nit-picky.

```lua
local _, instance = next(conf.softwire_config.instance)
local queue = instance.queue -- Or simply use it as instance and not define queue
```